index.ts

```diff
@@ -18,8 +18,48 @@ export class Tree<TValue extends number> {
   }
 
   private _insert(node: ITreeNode<TValue>, value: TValue) {
-    const direction = value >= node.value ? 'right' : 'left'
-    if (node[direction] !== undefined) this._insert(node[direction]!, value) // eslint-disable-line @typescript-eslint/no-non-null-assertion
+    const direction = this._setDirection(node, value)
+    /* eslint-disable */
+    if (node[direction] !== undefined) this._insert(node[direction]!, value)
     else node[direction] = { value }
+    /* eslint-enable */
+  }
+
+  private _setDirection(node: ITreeNode<TValue>, value: TValue) {
+    return value >= node.value ? 'right' : 'left'
+  }
+
+  public remove(value: TValue) {
+    if (this.root === undefined) throw new Error('There is no tree currently')
+    /* eslint-disable */
+    if (value === this.root!.value!) {
+      /* eslint-enable */
+      this.root = undefined
```

Comment:
More semantically correct is `null`

Reply:
it should be undefined as defined in type(either undenfined or ITreeNode<TValue>)